### DISCUSSION TITLE
Disable code navigation

### DISCUSSION THREAD

**Discussion by rodriguez-facundo** [community/community#17104] [Other Feature Feedback, Questions, & Ideas]
Would be great to be able to disable code navigation

**Comment by look** [comment]
@rodriguez-facundo thanks for the feedback. Can you tell us more about your use case? When and why do you want to disable code navigation?

**Reply by rodriguez-facundo** [reply]
Hi 👋, sometimes I read a function and I want to see (within the same page) where that function is been used. 
I used to double-click the function name, press `ctl+F` and then just hit `Enter` and I would be moving to each of the places where the function appears. But after Code Navigation became the default, it is not possible to double-click the function name. The code navigation pop-up will show up and when trying to use the mouse to carefully select the function name, it is not so simple, because you end up selecting more than needed, so then you need to delete the things that you selected in excess.

**Reply by hughetop** [reply]
I was just about to post the same thing. It would be great to be able to disable code navigation per repository. The popup is nice to see where/how variables or functions are defined, but it doesn't give enough context. So if I want to see more, I have to click on the definition, which goes to some weird blob (seems to be the latest commit) instead of just the default branch. Also, imported modules almost never work, especially from a different repository. If anything, an import will just reference itself, which is not useful. Plus the on-click behavior interferes with the text highlighting as mentioned above. It causes more frustration than it is helpful, so I'd rather just disable it.

**Reply by colinwm** [reply]
Just to clarify, are you referring to cs.github.com (new code search), or github.com?

**Reply by rodriguez-facundo** [reply]
Hi @colinwm 👋, just to this

<img width="658" alt="Screenshot 2022-05-23 at 4 31 24 PM" src="https://user-images.githubusercontent.com/32278395/169892722-3ac44876-adf6-47f3-8eab-9105f0938a08.png">

when hovering the function name, the pop-up shows instead of allowing me to select the string to do `ctl+F`. So, what I need to do next is to use the pad to move the cursor to the right place and then drag until I get the full string selected. before it was just a double-click and no pad required

**Reply by colinwm** [reply]
Got it! That absolutely makes sense

**Reply by Rosswell** [reply]
i second this request - i often find myself wanting to copy code from PRs to do a variety of things, including:

- test the validity of a suggestion im making in some scratch environment
- reformat some code in my IDE, also for the purposes of making suggestions
- general code navigation on a different branch - often ill want to see how something was being used prior to the PR changes that are being proposed. even if its in the same branch though, i dont want to be navigated away from the PR in order to derive context for something, and the definition/references aren't nearly enough to figure anything out about what im looking at, ive always needed to actually go to the source/destination file. additionally, navigating code in this way is very slow - between the choice of instant navigation in an IDE and having to load a webpage every time ill always pick the former

**Reply by hughetop** [reply]
"having to load a webpage every time"
That's what I really don't understand, a lot of the time the class/function/variable definition is in the same file. But instead of just navigating to the correct line with the "#L123" url syntax, it redirects to the latest commit, forcing it to load the exact same page contents again. Combined with how often it just doesn't display anything or only references itself, I'd rather just disable it.

**Reply by bef178** [reply]
I understand the problem here is not about an option to disable/enable code navigation, but the default behavior of double-click on code.

The default operation for double-click (I think) should be making the word under cursor selected. That behavior is shared among most IDEs.

The code navigation is good, but should be bound to ctrl-click or other shortcut.

**Reply by hughetop** [reply]
No, I want to disable it. It's more than just issues highlighting text. The definition it gives you (if it gives you anything) doesn't show enough context. If I click on it to see more context, it redirects to the latest commit page instead of the default branch. If the definition is in the same file, which it often is, that means it will reload the same page contents unnecessarily. It rarely, if ever, works on python imports, it'll just reference itself, which is not helpful. So I would rather disable it because it just causes more frustration.

**Reply by lzanolcastanheira** [reply]
This feature should remain optional until proven more stable and provides the user with the ability to enable/disable it (maybe as an experimental feature). It could also be improved like suggested above by adding some modifier keys (i.e. CTRL, CMD, ALT, etc.) to trigger it, prevent refreshes when on the same page, etc.

It's a nice feature but right now, it just makes code reading and PR reviewing more painful.

**Reply by mvasilkov** [reply]
Yes please allow us to disable this, it'll be the single greatest productivity boost if this garbage popup goes away.

**Reply by youurayy** [reply]
Double-clicking a word and then pressing Cmd/Ctrl+F nicely highlights the occurrences in Chrome. This is useful.

However the double-click selection is thwarted by Github's code-navigation popup, which doesn't even select the occurrences on the current page.

Additionally, in Python, most of the time the popup has no clue where the reference comes from and so it is usually empty. Utility value: negative.

**Reply by hughetop** [reply]
Good point, the fact that it takes over the on-click behavior _and_ doesn't even work properly makes it a hindrance and reduces overall usability. If it was one _or_ the other, it could possibly be redeemed.

**Reply by colinwm** [reply]
I want to update everyone on this. 

We're making significant changes to code navigation, including eliminating this popup. Stay tuned for GitHub Universe (in about 1 month), where we'll be launching these improvements. Once it launches, I'd love to hear your feedback about the new experience.

**Reply by smtchahal** [reply]
My use case: I just want to copy the function name, so I can search it in my IDE. But now every time I double click a function, instead of the text getting selected so I can copy it, the definition thing pops up, which is not useful to me, as I'll be navigating in my IDE.

**Reply by lzanolcastanheira** [reply]
@colinwm If I understood correctly, as per your last comment above you'd be launching these improvements by mid-November. Any estimates on that? Thanks.

**Reply by colinwm** [reply]
If you sign up for the new code search and code navigation beta (https://github.com/features/code-search), you'll get the new UI which doesn't have the popup.

And by the way, if you get access to that beta, you can double click function names to select them.

**Reply by hughetop** [reply]
@colinwm does code search and/or navigation work across internal repositories within a private organization?

**Reply by colinwm** [reply]
Yes (although cross-repo code navigation only works for Python)

**Reply by hughetop** [reply]
Can that be limited to repositories belonging to a specific team within the organization?

**Reply by colinwm** [reply]
You can search within a subset of repositories by creating a saved search, try typing `saved:` into the search bar and choosing "create saved search".

**Reply by lzanolcastanheira** [reply]
@colinwm That works! Thank you for the prompt response!

**Reply by toidiu** [reply]
I am seeing this behavior as or recently. Cant tell if there is a way to disable this and/or when it will be fixed by default. The link to beta code navigation doesnt allow you to signup for anything.

To clarify, my ask is to hide the code navigation menu and never show it. It takes up screen real-estate, counts towards the "find" count and makes it more difficult to navigate code. I prefer the Ctrl+F since I can use the keyboard to navigate rather than the mouse.

**Reply by toidiu** [reply]
This discussion is becoming an echo chamber of how this new feature is really annoying and I think its stemming from the fact that its quite distracting and a very different UI. I am curious if you are also receiving positive reviews. I do have to admit that after a month, the feature has become a bit more useful and a bit more second nature (nicely done devs). It might still be best to allow for disabling the feature.. maybe??

Has anyone else here also have a change of heart?

**Reply by mvasilkov** [reply]
@toidiu absolutely not, the thing is every bit as infuriating. Good job product people, you've made the world slightly worse.

**Reply by lzanolcastanheira** [reply]
If Product really wants this feature, please make it possible to enable/disable it. As a software engineer who daily and heavily uses Git/Github, I expect a quick and simpler interface when browsing code from a web browser, I just need to quickly see the code and files, not interact with it. If I need to do something more (e.g. navigate the code flow, highlight variables by name and scope, check class members of an instance, etc.), I open the repo on a proper IDE which is far more powerful and specialized (or even `github.dev`, which is very neat).

On top of that, the new Symbol panel also introduces unnecessary distraction and has no way to get disabled either. Even if closed, when you click somewhere on the code, it opens up again.

All in all, these things hinder simple tasks like reading a code or simply selecting some text to copy/paste somewhere else.

**Comment by qianbinbin** [comment]
I'm not sure if it's related to code navigation, but when you open a big file on github, double click to select a word, then ctrl+c or right click just stuck.
About 30 seconds later chrome ask you whether to kill the page. Finally the right click menu show up. It really sucks.

**Reply by AdamShwert** [reply]
Could you provide some more details for us? Does the page hang on a double click, or when you try to copy/right click? Also, how large of a file are we talking about? Thanks!

**Reply by qianbinbin** [reply]
For example, open this link https://github.com/privacy-protection-tools/anti-AD/blob/master/anti-ad-domains.txt , double click a domain to select, then right click, it'll be stuck for several seconds until the menu appears. During this period we can do nothing but wait.
The larger a file is, the more time to wait.
Tested on macOS 10.15.7, Chrome 113.0.5672.126

**Reply by AdamShwert** [reply]
I'm not seeing this behavior when I try to replicate - does it happen in a different browser too?

**Reply by qianbinbin** [reply]
Safari behaves in a different way. It takes time to double click to select, after right click, the menu appears immediately. It looks more smooth than Chrome, however still slower than ordinary pages on Safari.

**Reply by AdamShwert** [reply]
If you copy the contents of the file and paste it in an input box like this comment box, does the same thing occur?

**Reply by qianbinbin** [reply]
I recorded a video here:
https://github.com/community/community/assets/7856168/d07a9c8a-5e49-4011-a0e8-9b881ec42c24

**Reply by qianbinbin** [reply]
> If you copy the contents of the file and paste it in an input box like this comment box, does the same thing occur?

Yes. But when I paste on other sites like google translate, it runs smoothly.

**Reply by AdamShwert** [reply]
Okay, thank you for the video and the report. I'll see what we can do

**Comment by danuker** [comment]
I dislike the distraction when moving my mouse around (the highlighting, and the delayed animation when double-clicking).

If you want me to ever use this feature, it should only be active when I hold `Ctrl`.

**Reply by danuker** [reply]
I have disabled the highlighting by adding the following in my `userContent.css`:

```css
@-moz-document domain(github.com) {
    .pl-token:hover, .pl-token.active {
        cursor: auto!important;
        background: none!important;
    }
}
```

I have disabled the popover with the following uBlock Origin filter:

`github.com##.js-tagsearch-popover.anim-scale-in.Popover`
